### PR TITLE
add SubmitTxsDayEpoch, SubmitTxsStrideEpoch, 10 min timeout to claims

### PR DESCRIPTION
**Pull request by asalzmann**
**Summary**
Add timeouts to ICA calls. Specifically, in `msg_server_submit_tx.go`
- Add `SubmitTxsDayEpoch` function
- Add `SubmitTxsStrideEpoch` function
- Add 10 min timeout to `SubmitTxs` called in `ClaimUndelegatedTokens`

This is necessary because our record keeping logic assumes packets time out before the next epoch, to avoid retrying transfers / stakes that are already being processed.

**Test plan**
Run 1.sh, 2.sh, 3.sh, 4.sh

### CODE REVIEW

File: x/stakeibc/keeper/msg_server_claim_undelegated_tokens.go [hunk 1]
```diff
@@ -52,7 +53,9 @@ func (k msgServer) ClaimUndelegatedTokens(goCtx context.Context, msg *types.MsgC
 		Amount:      sdk.NewCoins(sdk.NewInt64Coin(userRedemptionRecord.Denom, int64(userRedemptionRecord.Amount))),
 	})
 	
-	sequence, err := k.SubmitTxs(ctx, hostZone.GetConnectionId(), msgs, *redemptionAccount)
+	// Give claims a 10 minute timeout
+	timeout := uint64((10 * time.Minute).Nanoseconds())
```
Comment (riley-stride) [inline x/stakeibc/keeper/msg_server_claim_undelegated_tokens.go:57]: should we make this a param?

File: x/stakeibc/keeper/callbacks.go [hunk 2]
```diff
@@ -144,7 +144,7 @@ func WithdrawalBalanceCallback(k Keeper, ctx sdk.Context, args []byte, query icq
 	ctx.Logger().Info(fmt.Sprintf("Submitting withdrawal sweep messages for: %v", msgs))
 
 	// Send the transaction through SubmitTx
-	_, err = k.SubmitTxs(ctx, zone.ConnectionId, msgs, *withdrawalAccount)
+	_, err = k.SubmitTxs(ctx, zone.ConnectionId, msgs, *withdrawalAccount, ^uint64(0) >> 1)
```
[the remark this answers is not in the document]
Comment (riley-stride) [reply x/stakeibc/keeper/callbacks.go:147]: note in suggestion above I removed ">>1"

[the remark this answers is not in the document]
Comment (riley-stride) [reply x/stakeibc/keeper/callbacks.go:147]: why no timeouts for ICA bank sends from withdrawal => delegation account?

[the remark this answers is not in the document]
Comment (asalzmann) [reply x/stakeibc/keeper/callbacks.go]: Added timeouts! Sorry I think I lost a commit here

File: x/stakeibc/keeper/msg_server_claim_undelegated_tokens.go (hunk 1 above)
Comment (asalzmann) [reply x/stakeibc/keeper/msg_server_claim_undelegated_tokens.go]: Good idea, done.

File: x/stakeibc/keeper/msg_server_submit_tx.go [hunk 3]
```diff
@@ -170,8 +171,41 @@ func (k Keeper) UpdateWithdrawalBalance(ctx sdk.Context, zoneInfo types.HostZone
...
+}
+
+func (k Keeper) SubmitTxsEpoch(ctx sdk.Context, connectionId string, msgs []sdk.Msg, account types.ICAAccount, epochType string) (uint64, error) {
+	epochInfo, found := k.epochsKeeper.GetEpochInfo(ctx, epochType)
+	if !found {
+		k.Logger(ctx).Error("Failed to get epoch info for %s", epochType)
+		return 0, sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "Failed to get epoch info for %s", epochType)
+	}
+	// BUFFER by 5% of the epoch length
+	BUFFER := epochInfo.GetDuration().Nanoseconds() / 20
```
Comment (vish-stride) [inline x/stakeibc/keeper/msg_server_submit_tx.go:197]: should we make the buffer (e.g. 20) a param?

**Review by vish-stride — APPROVED** [review]
looks great!

File: x/stakeibc/keeper/msg_server_submit_tx.go (hunk 3 above)
Comment (asalzmann) [reply x/stakeibc/keeper/msg_server_submit_tx.go]: Done.

**Pull request merged**